src/main/timeUtils.ts

```diff
@@ -7,12 +7,24 @@ export const integerDivide: (a: number, b: number) => number = (a, b) => {
 };
 
 // 00:00:00:00
-export const secondToTimestamp: (num: number) => string = (num) => {
+export const secondToEDLTimestamp: (num: number, fps: number) => string = (
+  num,
+  fps
+) => {
   if (num < 0) {
     throw new Error('Negative Input');
   }
+  const edlFps = Math.max(fps, 30);
 
-  return new Date(num * 1000).toISOString().slice(11, 22).replace('.', ':');
+  const date = new Date(num * 1000);
+  const frame = Math.floor((edlFps * date.getMilliseconds()) / 1000);
+  /**
+   * For the frame gap problem, making timestamps round down 1 frame while having the
+   * input duration round 1 up frame fills in microgaps cleanly.
+   */
+
+  const timestamp = date.toISOString().slice(11, 20).replace('.', ':');
```

Comment:
What is this doing? Looks a bit hacky and I feel like it might be fragile (e.g. if some component of the date takes up an extra digit in some edge case then it will be off by one)

Reply:
Changed it to be clear that it converts to a date for ease of parsing hours, mins, sec, etc.